tests/ops/op_math/test_composite.py

```diff
@@ -236,6 +236,18 @@ def test_tensor_and_hamiltonian_converted(self):
 class TestMscMethods:
     """Test dunder and other visualizing methods."""
 
+    @pytest.mark.parametrize("math_op", [qml.prod, qml.sum])
+    @pytest.mark.parametrize("method", ["terms", "eigvals", "__copy__", "label"])
+    def test_recursion_depth_error_message(self, math_op, method):
+        """Tests that a sensible error is raised from a recursion error"""
+
+        op = qml.PauliRot(np.random.uniform(0, 2 * np.pi), "X", wires=1)
+        for _ in range(2000):
+            op = math_op(op, qml.PauliRot(np.random.uniform(0, 2 * np.pi), "Y", wires=1))
```

Comment:
Given creating this huge op might be a bit more expensive to create than normal, maybe we should create it once, and then loop over all the different methods, instead of having to create it once per attribute we test?

Reply:
Done